docker/datahub-ingestion-base/Dockerfile

```diff
@@ -50,24 +58,49 @@ RUN apt-get update && apt-get upgrade -y \
     ldap-utils \
     unixodbc \
     libodbc2 \
-    && python -m pip install --no-cache --upgrade pip 'uv>=0.1.10' wheel setuptools \
     && apt-get clean \
     && rm -rf /var/lib/{apt,dpkg,cache,log}/
 
+# Set Python 3.10 as the default python3
+RUN update-alternatives --install /usr/bin/python3 python3 /usr/bin/python3.10 1 \
+    && update-alternatives --install /usr/bin/python python /usr/bin/python3 1
+
+# Optionally set corporate mirror for pip
+ARG PIP_MIRROR_URL
+RUN if [ "${PIP_MIRROR_URL}" != "https://pypi.python.org/simple" ] ; then pip config set global.index-url ${PIP_MIRROR_URL} ; fi
+ENV UV_INDEX_URL=${PIP_MIRROR_URL}
+
 COPY --from=powerman/dockerize:0.19 /usr/local/bin/dockerize /usr/local/bin
 
 COPY ./docker/datahub-ingestion-base/entrypoint.sh /entrypoint.sh
 
-RUN addgroup --gid 1000 datahub && \
-    adduser --disabled-password --uid 1000 --gid 1000 --home /datahub-ingestion datahub && \
+RUN existing_group=$(getent group 1000 | cut -d: -f1) && \
+    if [ -n "$existing_group" ]; then \
+        echo "Using existing group $existing_group with GID 1000"; \
+    else \
+        addgroup --gid 1000 datahub; \
+        existing_group="datahub"; \
+    fi && \
+    existing_user=$(id -nu 1000 2>/dev/null || echo "") && \
+    if [ -n "$existing_user" ]; then \
+        echo "Renaming existing user $existing_user to datahub"; \
+        usermod -l datahub -d /datahub-ingestion $existing_user; \
+    else \
+        adduser --disabled-password --uid 1000 --gid 1000 --home /datahub-ingestion datahub; \
+    fi && \
+    # Create and set proper permissions for datahub directories
+    mkdir -p /datahub-ingestion/.venv && \
```

Comment:
this shouldn't be necessary? if all subsequent commands use `USER datahub`, then things should work just fine right?